### ISSUE TITLE
Create builders for pub structs with lots of properties

### ISSUE THREAD

**Issue by valentinegb** [valentinegb/openai#16]
### Problem

Currently, there's a little bit of boilerplate when creating, for example, a `Completion`. (This boilerplate being `Default::default()`.) It doesn't look all that nice.

### Solution

We can get rid of boilerplate by introducing builders. This would also allow for more flexible ways of creating completions and such, like setting `echo` later in code if a certain condition is met. With builders, creating a `Completion` would look a bit like this:

```rs
let completion = Completion::new(ModelID::TextDavinci003)
    .prompt(“Say this is a test”)
    .max_tokens(7)
    .temperature(0.0)
    .create()
    .await
    .unwrap()
    .unwrap();
```

**Comment by valentinegb** [comment]
Postponed until #18 is done. Field requirements are already enforced by the API, so it's redundant to enforce them twice. Once #18 is done, builders can be more easily made with the [derive_builder](https://crates.io/crates/derive_builder) crate.

**Comment by valentinegb** [comment]
#18 has been merged! Work on this is being restarted.
